### PR TITLE
Initial calendar implementation

### PR DESCRIPTION
**Pull request by TimaQT**
Initial calendar setup completed, including the interface and fronted connection to the backend, along with a basic template for the assignment reader. 

Note: _The full integration with PCSS will come later (RESTful API and integration). The code structure may change depending on the API PCSS implementation. Features like Jump To Date, assignment displays inside the calendar cells and popups will also be added in the next phase._

# Pull Request for JIRA Ticket: JASPER-71

## Issue ticket number and link  
https://jag.gov.bc.ca/jira/browse/JASPER-71

## Description

Initial calendar setup completed, including the interface and fronted connection to the backend, along with a basic template for the assignment reader. The code structure may change depending on the API PCSS implementation.

Note: _The full integration with PCSS will come later (RESTful API and integration). Features like Jump To Date, assignment displays inside the calendar cells and popups will also be added in the next phase._

## Type of change
- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Manual local testing for hard-coded assignments display
- [x] Unit tests will also be added in the next phase.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules   

- [x] I have commented my code, particularly in hard-to-understand areas. More comments would be added during further calendar development
- [x] There is a new warning within the calendar.js library - the warning does not affect the functionality, however it would be also investigated.

### CODE REVIEW

**Comment by WadeBarnes** [comment]
@TimaQT, Have you reviewed the build errors here; https://github.com/bcgov/jasper/actions/runs/11672302245/job/32500602413?pr=29

Would you be able to provide some details on your local environment so we can determine what needs to be updated in the pipeline if needed?

Are you using the devContainer?
What versions of development tools and frameworks are you using?  e.g. Node and npm versions.  i.e. How does you're environment differ from the one defined in the `main` branch?

**Comment by TimaQT** [comment]
> @TimaQT, Have you reviewed the build errors here; https://github.com/bcgov/jasper/actions/runs/11672302245/job/32500602413?pr=29
> 
> Would you be able to provide some details on your local environment so we can determine what needs to be updated in the pipeline if needed?
> 
> Are you using the devContainer? What versions of development tools and frameworks are you using? e.g. Node and npm versions. i.e. How does you're environment differ from the one defined in the `main` branch? Please provide step by step instructions or how you're building the frontend for local testing.


_Are you using the devContainer?_

i'm opening folder locally. The issue is that I needed to install two libraries, FullCalendar and MonthPicker, but I wasn’t able to use npm within the development container.


_What versions of development tools and frameworks are you using? e.g. Node and npm versions. i.e. How does you're environment differ from the one defined in the main branch?_
├── @vue/cli-service@5.0.8
├── vue-month-picker@1.7.2
└── vue-template-compiler@2.6.12
Node.js v20.16.0


_Please provide step by step instructions or how you're building the frontend for local testing._

- open project in a local folder
- run docker desktop
- open git bash terminal, 
- run  cd docker 
- run ./manage debug


I’m encountering the same error on my local machine, but it doesn’t prevent the calendar from functioning correctly. While waiting for the update from Vue 2 to Vue 3, I’ll try downgrading the calendar version to find one compatible with the current setup. This might result in some functions no longer working, but it’s worth exploring to see if there’s a version that works with Vue 2 while maintaining the current calendar behavior.

jasper-web-dev-1  | ERROR in /opt/app-root/src/node_modules/@fullcalendar/vue/dist/FullCalendar.d.ts(4,29):
jasper-web-dev-1  | 4:29 Generic type 'ExtendedVue' requires 5 type argument(s).
jasper-web-dev-1  |     2 | import { Calendar, CalendarOptions } from '@fullcalendar/core';
jasper-web-dev-1  |     3 | import { CustomRendering } from '@fullcalendar/core/internal';
jasper-web-dev-1  |   > 4 | declare const FullCalendar: import("vue/types/vue.js").ExtendedVue<Vue<Record<string, any>, Record<string, any>, never, never, (event: string, ...args: any[]) => Vue<Record<string, any>, Record<string, any>, never, never, any>>, {
jasper-web-dev-1  |       |                             ^
jasper-web-dev-1  |     5 |     renderId: number;
jasper-web-dev-1  |     6 |     customRenderingMap: Map<string, CustomRendering<any>>;
jasper-web-dev-1  |     7 | }, {

**Comment by WadeBarnes** [comment]
@TimaQT, It looks like the `web/package-lock.json` was corrupted during the recent merge.  It doesn't look like it's in a valid json format anymore.

**Review by WadeBarnes — APPROVED** [review]
Looks good for the first pass.  I think the next step, which should be done as an enhancement once this is merged, is to get JASPER actually wired to PCSS so we can start seeing some real data flow.

File: api/Models/Calendar/CalendarDay.cs [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        public bool showAM
+        {
+            get
+            {
+                return showPM;
+            }
+        }
+        public bool showPM
+        {
+            get
```
Comment (ronaldo-macapobre) [inline api/Models/Calendar/CalendarDay.cs:26]:
```suggestion
            get
            {
               return showPMLocation || (this.Assignment?.ActivityAm?.CourtRoomCode != this.Assignment?.ActivityPm?.CourtRoomCode)
                    || (this.Assignment?.ActivityAm?.ActivityDescription != this.Assignment?.ActivityPm?.ActivityDescription);
            }
```

**Pull request merged**